common/internal_model/src/main/scala/uk/ac/wellcome/models/parse/FreeformDate.scala

```diff
@@ -0,0 +1,103 @@
+package uk.ac.wellcome.models.parse
+
+import java.time.LocalDate
+
+import uk.ac.wellcome.models.work.internal.InstantRange
+
+/**
+  *  Models freeform text representing some date or period
+  *
+  *  @param date The date or date range
+  *  @param inferred Whether the date is known exactly or inferred
+  */
+case class FreeformDate(date: Either[FuzzyDate, DateRange],
```

Comment:
Feels like `FuzzyDate` & `DateRange` should share a common trait enabling you to disambiguate them without an `Either`.

Reply:
Sounds good